include/stout/flags/flags.cc

```diff
@@ -536,34 +536,15 @@ void Parser::Parse(const std::vector<ArgumentInfo>& values) {
       }
     }
 
-    // Parse the value using an overloaded parser if provided.
```

Comment:
Are you able to add a (unit) test that captures the bug that this PR fixes? A test would give me confidence that we won't risk regressing this fix.